extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/EventBusConsumer.java

```diff
@@ -175,10 +195,20 @@ private static void implementInvoke(BeanInfo bean, MethodInfo method, ClassCreat
         } else {
             // Parameter is payload
             ResultHandle bodyHandle = invoke.invokeInterfaceMethod(MESSAGE_BODY, messageHandle);
-            ResultHandle returnHandle = invoke.invokeVirtualMethod(
-                    MethodDescriptor.ofMethod(bean.getImplClazz().name().toString(), method.name(),
-                            method.returnType().name().toString(), paramType.name().toString()),
-                    beanInstanceHandle, bodyHandle);
+            ResultHandle returnHandle;
+            if (headerType.isPresent()) {
+                // TODO: Handle headers from both mutiny and non-mutiny? Will it always be mutiny?
+                ResultHandle headerHandle = invoke.invokeInterfaceMethod(MESSAGE_HEADERS, messageHandle);
+                returnHandle = invoke.invokeVirtualMethod(
+                        MethodDescriptor.ofMethod(bean.getImplClazz().name().toString(), method.name(),
+                                method.returnType().name().toString(), headerType.get().name().toString(), paramType.name().toString()),
+                        beanInstanceHandle, headerHandle, bodyHandle);
```

Comment:
I _think_ adding `headerHandle` before `bodyHandle` is correct?

Reply:
Yes, given the fact that we validate that headers must be first.